package.json

```diff
@@ -82,11 +83,13 @@
     "git-url-parse": "^13.1.0",
     "globby": "^11.0.4",
     "https-proxy-agent": "^5.0.1",
+    "is-docker": "^2.2.1",
```

Comment:
ahh, node ecosystem. Since this package has no other external dependencies let's pin an exact version and let one of the numerous bots notify us of any required updates.

Reply:
Is this package less trustworthy than any of the others? 🙃

Reply:
no, and that is my point. It has no dependencies and is unlikely to change so pin the exact version...

Reply:
It's unlikely to change but already on version 2.2.1 🤔

Reply:
Software is never finished, just abandoned

Reply:
as a security-minded person, I'm surprised you don't see the point I'm making as with other cases of other random packages, but whatever...

Reply:
I pride myself of seeing the point: having a watering hole less in the project.

My hesitation is still based on the fact that this is just a very small portion of our dependency tree and equally or less likely to suffer a supply chain attack than any of the other dependencies.

Lock files are meant to pin dependencies as you intend. Unless my Node.js notions are wrong (thing I don't discard), the `package.json` file has another purpose.

Reply:
I've seen our lock file be rebuilt multiple times from scratch which is based on the package.json file, so all it would take is some unfortunate timing.

I agree it is a small proportion so why should we even allow the possibility when there is nothing to be gained?[^1]

[^1]: nothing wrong with that; perhaps suggest to him that as a very prolific open source developer he sign commits his commits and tags?

Reply:
> _I've seen our lock file be rebuilt multiple times from scratch which is based on the package.json file, so all it would take is some unfortunate timing._

I'd argue that it's partly because of some development “practices” like running `npm install` instead of `npm ci` and committing unsupervised lock file changes on every single pull request; and partly because the Node.js package ecosystem is rather “optimistic” to put it mildly.

> _I agree it is a small proportion so why should we even allow the possibility when there is nothing to be gained?_

We'd gain consistency. 🤷🏼‍♂️

Reply:
Anyway, resolved with 4880fd7

Reply:
Should we ask #javascript for more opinions on the topic? It would be nice to do this always if we find it's a good idea, or never if we don't.